### PR TITLE
Log agent errors using write_run_logs

### PR DESCRIPTION
**Pull request by zdhughes**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?

This PR updates the agent to log errors using `write_run_logs` rather than the deprecated `write_run_log`. 

## Why is this PR important?

While we are in fact only logging one thing, presenting a deprecation warning right alongside an agent error confounds the messaging. This moves us away from a deprecated API call and hopefully improves UX.

### CODE REVIEW

File: src/prefect/agent/agent.py [hunk 1]
```diff
@@ -360,11 +360,15 @@ def _log_flow_run_exceptions(self, flow_runs: list, exc: Exception) -> None:
...
             )
-            self.client.write_run_log(
-                flow_run_id=flow_run.id,  # type: ignore
-                name="agent",
-                message=str(exc),
-                level="ERROR",
+            self.client.write_run_logs(
+                [
+                    dict(
+                        flow_run_id=flow_run.id,  # type: ignore
```
Comment (cicdw) [inline src/prefect/agent/agent.py]:
```suggestion
                        flowRunId=flow_run.id,  # type: ignore
```

Comment (cicdw) [reply src/prefect/agent/agent.py]: The schema changed a little bit for this API call

**Comment by cicdw** [comment]
Test failures are unrelated to this PR - there's a brittle test that I need to change.

**Comment by zdhughes** [comment]
Good stuff-- thanks for the catch.

File: tests/agent/test_agent.py [hunk 2]
```diff
@@ -343,9 +343,9 @@ def test_agent_logs_flow_run_exceptions(monkeypatch, runner_token):
         exc=ValueError("Error Here"),
     )
 
-    assert client.write_run_log.called
-    client.write_run_log.assert_called_with(
-        flow_run_id="id", level="ERROR", message="Error Here", name="agent"
+    assert client.write_run_logs.called
+    client.write_run_logs.assert_called_with(
+        [dict(flow_run_id="id", level="ERROR", message="Error Here", name="agent")]
```
Comment (cicdw) [inline tests/agent/test_agent.py]: This test needs to be changed also 👍

**Pull request merged**